DataSource/Util/ChoicesBuilder.php

```diff
@@ -22,39 +22,72 @@ public function __construct(EntityManager $entityManager)
     /**
      * @param array|callable $choicesConfiguration
      *
-     * @return array|null
+     * @return array
      *
      * @throws \Exception
      */
     public function build($choicesConfiguration)
     {
-        $choices = null;
-
         if (
             is_array($choicesConfiguration) &&
-            isset($choicesConfiguration['repository']) &&
-            isset($choicesConfiguration['field'])
+            isset($choicesConfiguration['repository'])
         ) {
-            if ($choicesConfiguration['repository'] instanceof EntityRepository) {
-                $repository = $this->getSpecifiedRepository($choicesConfiguration['repository']);
-            } else {
-                $repository = $this->entityManager->getRepository($choicesConfiguration['repository']);
-            }
-            $choicesField = $choicesConfiguration['field'];
-            $choicesSort = isset($choicesConfiguration['sort']) ? $choicesConfiguration['sort'] : null;
-            $choices = $this->getChoicesFromRepository($repository, $choicesField, $choicesSort);
-        } elseif (is_callable($choicesConfiguration)) {
-            // Choices is a callable. It will generate the choices array.
-            $choices = $choicesConfiguration();
-            if (!is_array($choices)) {
-                throw new \Exception("Choices callback defined in table configurations must return arrays");
-            }
-        } elseif (is_array($choicesConfiguration)) {
-            // Just a plain array. Use it as choices
-            $choices = $choicesConfiguration;
+            return $this->getChoicesFromRepository($choicesConfiguration);
+        }
+        if (is_callable($choicesConfiguration)) {
+            return $this->getChoicesFromCallable($choicesConfiguration);
+        }
+        if (is_array($choicesConfiguration)) {
+            return $choicesConfiguration;
         }
 
-        return $choices;
+        throw new \Exception('No usable configuration was found');
+    }
+
+    /**
+     * @param array $choicesConfiguration
+     *
+     * @return array
+     *
+     * @throws \Exception
+     */
+    private function getChoicesFromRepository(array $choicesConfiguration)
+    {
+        if ($choicesConfiguration['repository'] instanceof EntityRepository) {
+            $repository = $this->getSpecifiedRepository($choicesConfiguration['repository']);
+        } else {
+            $repository = $this->entityManager->getRepository($choicesConfiguration['repository']);
+        }
+
+        if (isset($choicesConfiguration['field'])) {
```

Comment:
If `field` and `method` are set then field always wins. Is this desired? or should we handle it or throw an exception?